### DISCUSSION TITLE
UX improvement: show how much hard disk space was saved

### DISCUSSION THREAD

**Discussion by nikoladev** [pnpm/pnpm#3173] [Ideas]
During the installation of packages the CLI shows how much packages were reused and downloaded:

```bash
Resolving: total 1123, reused 1100, downloaded 23
```

I think this should also show how much MBs were saved on disk because the packages weren't copied to the `node_modules` folder:

```bash
Resolving: total 1123, reused 1100, downloaded 23
Saved 88.7 MBs of disk space!
```

This is a unique selling point that pnpm has and it should be highlighted. Especially because this will be shown after every package install (again and again) and will give the developer a positive feeling linked to pnpm, helping pnpm's reputation.

**Comment by zkochan** [comment]
I like this idea. But only if it won't require additional operations that would slow down installation.

Currently, we have the info about each file's size in the package's [integrity.json](https://github.com/pnpm/sample-project/blob/master/pnpm-store/2/registry.npmjs.org/chalk/1.1.3/integrity.json)

**Comment by zkochan** [comment]
Also, it will show only the minimal amount of saved disk space. Sometimes npm/yarn copy/paste one package more than 1 time.

Another good idea might be if the saved space is more than some amount (500 MB or so), we can add a twitter link that suggests users tweet something like: `@pnpmjs just saved 945 MB on my hard drive!`

**Reply by privatenumber** [reply]
This sounds like a great idea!
Are there any updates on this?

I think a lot of users don't realize how much pnpm is saving them. If it was reported on every `pnpm install` or if there was a command to check (eg. `pnpm savings` or `pnpm stats`), I think adoption rates will skyrocket!

I wouldn't add a Twitter link in a terminal output though. I think engineers impressed with it will tweet screenshots of it, which is even more engaging.

**Comment by nikoladev** [comment]
> I like this idea. But only if it won't require additional operations that would slow down installation.

As long as we don't do any measurements during install and only show the result at the end the performance shouldn't be a problem.

> Also, it will show only the minimal amount of saved disk space. Sometimes npm/yarn copy/paste one package more than 1 time.

Is there any way we could measure this? Otherwise we can just say that `pnpm saved xx MB (and probably more!)` and provide a link to a page where they can read the specifics of how pnpm works.

**Comment by patoliadixit** [comment]
Any update on this feature?

**Reply by zkochan** [reply]
no
